network/network.go

```diff
@@ -369,30 +394,48 @@ func (n *Network) connectToKnownNodes(nodeDID did.DID) error {
 		return err
 	}
 	for _, node := range otherNodes {
-		if !nodeDID.Empty() && node.ID.Equals(nodeDID) {
-			// Found local node, do not discover.
-			continue
-		}
-	inner:
-		for _, service := range node.Service {
-			if service.Type == transport.NutsCommServiceType {
-				var nutsCommUrl transport.NutsCommURL
-				if err = service.UnmarshalServiceEndpoint(&nutsCommUrl); err != nil {
-					log.Logger().
-						WithError(err).
-						WithField(core.LogFieldDID, node.ID.String()).
-						Warn("Failed to extract NutsComm address from service")
-					continue inner
-				}
+		n.connectToDID(nodeDID, node)
+	}
+
+	// Assume this is a new node when no NutsComm endpoints are found.
+	n.assumeNewNode = len(otherNodes) == 0
```

Comment:
`connectToDID()` uses this field, which is called above, so shouldn't it be set before the loop?

(indicates a missing test?)

Reply:
if `len(otherNodes) == 0` the loop is ran 0 times.
Will move it up to avoid confusion